### PR TITLE
[CMakeList.txt] add the option to compile with or without Fortran

### PR DESCRIPTION
**Pull request by vacary**
This PR tries to answer to the issue #22

### CODE REVIEW

**Comment by radarsat1** [comment]
This works well for me.  Also, allows to build with `-g Ninja`!  (Actually I had to make one small change that I will submit.)  Any reason to be wary of merging it?  I don't see anything dangerous.  @xhub?

I would add to the documentation a clear list of things (solvers..) that will not work with the flag disabled.

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +39,7 @@ if(USER_OPTIONS_FILE)
   include(${input_file})
   # copy user file to binary dir, just for 'visual' check.
   configure_file(${input_file} ${input_file} COPYONLY)
+  assert(WITH_FORTRAN "The option WITH_FORTRAN is not defined in the user options file")
 else()
```
Comment (xhub) [inline CMakeLists.txt]: Why do we need that?

File: control/control_tests.cmake [hunk 2]
```diff
@@ -15,8 +15,11 @@ if(WITH_${COMPONENT}_TESTING)
   # Simulation tests
   BEGIN_TEST(src/tests)
   
-  NEW_TEST(tests PIDTest.cpp SMCTest.cpp ObserverTest.cpp)
+  NEW_TEST(tests)
   
+  IF(WITH_FORTRAN)
+    NEW_TEST(PIDTest.cpp SMCTest.cpp ObserverTest.cpp)
+  ENDIF()
   END_TEST()
```
Comment (xhub) [inline control/control_tests.cmake]: This is a bit misleading: it should be WITH_EVENTDRIVEN. We only have fortran code that perform an event-driven integration. If we have SUNDIALS bindings, then we can use the latter for performing ED integration.

Ideally I would switch inside the test to a TS OSI if no ED OSI are available.

File: control/swig/CMakeLists.txt [hunk 3]
```diff
@@ -23,4 +23,8 @@ set(${COMPONENT}_SWIG_DEFS "${SWIG_DEFS}")
 
 include(swig_python_tools)
 swig_module_setup(${COMPONENT}_PYTHON_MODULES)
-build_python_tests()
\ No newline at end of file
+if(NOT WITH_FORTRAN)
+  list(APPEND ${COMPONENT}_python_excluded_tests
+    ${CMAKE_CURRENT_SOURCE_DIR}/tests/test_smc.py)
+endif()
+build_python_tests()
```
Comment (xhub) [inline control/swig/CMakeLists.txt]: ditto WITH_EVENTDRIVEN

File: kernel/kernel_tests.cmake [hunk 4]
```diff
@@ -76,9 +76,12 @@ if(WITH_${COMPONENT}_TESTING)
...
 
-  NEW_TEST(testSimulationTools ZOHTest.cpp OSNSPTest.cpp)
-
+  NEW_TEST(testSimulationTools  OSNSPTest.cpp)
 
+  IF(WITH_FORTRAN)
+    NEW_TEST(ZOHTest.cpp)
+  ENDIF()
+  
   END_TEST()
```
Comment (xhub) [inline kernel/kernel_tests.cmake]: ditto WITH_EVENTDRIVEN

File: kernel/swig/CMakeLists.txt [hunk 5]
```diff
@@ -29,6 +29,11 @@ if(NOT WITH_XML)
   list(APPEND ${COMPONENT}_python_excluded_tests
     ${CMAKE_CURRENT_SOURCE_DIR}/tests/test_xml.py)
 endif()
+if(NOT WITH_FORTRAN)
+  list(APPEND ${COMPONENT}_python_excluded_tests
+    ${CMAKE_CURRENT_SOURCE_DIR}/tests/test_matrix_exp.py)
+endif()
+
 build_python_tests()
```
Comment (xhub) [inline kernel/swig/CMakeLists.txt]: ditto WITH_EVENTDRIVEN

**Review by xhub — APPROVED** [review]
It's a good idea to have this flag. I would like to see a WITH_EVENTDRIVEN option to control whether some control tests are performed.

**Comment by vacary** [comment]
I think with have to separate WITH_FORTRAN form WITH_ EVENTDRIVEN since we will implement sundials inregration

**Pull request merged**